### PR TITLE
refactor: the way of plugin initialize load

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind improvement
/area core
/milestone 2.0

#### What this PR does / why we need it:
1. 优化插件初始化加载方式及 Plugin 自定义模型资源的更新
2. 插件 plugin.yaml 中 license 配置不再支持只配置字符串，而使用如下替代
```yaml
license:
  - name: "MIT"
```
3. 可以在 application.yaml 中配置
```yaml
halo:
  initial-extension-locations:
    - "path/to/extensions/yaml"
```
用于在系统启动时创建或更新自定义模型数据
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
/cc @halo-dev/sig-halo 
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by ruibaby — COMMENTED** [review]
/lgtm

**Review by ruibaby — COMMENTED** [review]
/lgtm

File: src/main/java/run/halo/app/infra/properties/HaloProperties.java [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 /**
  * @author guqing
- * @date 2022-04-12
+ * @since 2022-04-12
  */
+@Data
 @ConfigurationProperties(prefix = "halo")
 public class HaloProperties {
 
+    private Set<String> initializeExtensionLocations;
```
Comment (JohnNiang) [inline src/main/java/run/halo/app/infra/properties/HaloProperties.java:15]:
```suggestion
    private Set<String> initialExtensionLocations;
```

**Review by JohnNiang — CHANGES_REQUESTED** [review]

**Review by JohnNiang — APPROVED** [review]
/lgtm

**Pull request merged**